engine/src/main/java/org/destinationsol/common/SolMath.java

```diff
@@ -67,146 +90,185 @@ public static float approach(float src, float dst, float speed) {
     }
 
     /**
-     * Same as {@code approach()}, but in the radial coordinates. That is, 170 degrees would move to -10 by growing through 175 degrees
+     * Use this method when you want some angle to gradually transform to desired angle.
+     *
+     * @param src   Current angle in degrees
+     * @param dst   Desired angle in degrees
+     * @param speed Speed of change in degrees per call
+     * @return New angle closer to the desired one
      */
     public static float approachAngle(float src, float dst, float speed) {
-        float diff = norm(dst - src);
-        float da = abs(diff);
-        if (da <= speed) {
+        if (angleDiff(src, dst) <= speed) {
             return dst;
         }
-        return diff > 0 ? src + speed : src - speed;
+        float diff = norm(dst - src);
+        return norm(diff > 0 ? src + speed : src - speed);
     }
 
     /**
-     * Normalizes the angle
+     * Normalizes the angle, ie puts it in the range [-179.999, 180].
```

Comment:
Technically, it puts it in the range `(-180, 180]`.

Reply:
Technically, in czech at least, it puts it in range `(-180, 180>`. Didn't know the right notation, so I had to help myself in any way I could.